rct229/rulesets/ashrae9012019/ruleset_functions/compare_swh_dist_systems_and_components.py

```diff
@@ -0,0 +1,133 @@
+    compare_context_str: str
+        compare context value from the extra schema (e.g., RMD Test, AppG Used By TCDs, AppG P_RMD Equals U_RMD, AppG B_RMD Equals P_RMD, AppG B_RMDs Same)
+    swh_distribution_id: str
+        id of the `service_water_heating_distribution_systems` key to compare
+
+    Returns
+    -------
+    errors: list
+        An array of errors encountered in this function and in compare_context_pair. If errors is an empty array, then all elements that were expected to match do match
+
+    """
+
+    errors_list = []
+    rmd1_swh_system_and_equip_dict = (
+        get_swh_equipment_associated_with_each_swh_distribution_system(rmd1)
+    )
+    rmd2_swh_system_and_equip_dict = (
+        get_swh_equipment_associated_with_each_swh_distribution_system(rmd2)
+    )
+
+    # service_water_heating_distribution_systems
+    assert_(
+        rmd1_swh_system_and_equip_dict[swh_distribution_id],
+        f"SWH Distribution System {swh_distribution_id} not found in the `rmd1`.",
+    )
+    assert_(
+        rmd2_swh_system_and_equip_dict[swh_distribution_id],
+        f"SWH Distribution System {swh_distribution_id} not found in the `rmd2`.",
+    )
+    rmd1_swh_equipment_dict = rmd1_swh_system_and_equip_dict[swh_distribution_id]
+    rmd1_swh_distribution = find_exactly_one_service_water_heating_distribution_system(
+        rmd1, swh_distribution_id
+    )
+    rmd2_swh_equipment_dict = rmd2_swh_system_and_equip_dict[swh_distribution_id]
+    rmd2_swh_distribution = find_exactly_one_service_water_heating_distribution_system(
+        rmd2, swh_distribution_id
+    )
+
+    index_content_dict, compare_context_dict = (
+        (rmd1_swh_distribution, rmd2_swh_distribution)
+        if len(rmd1_swh_distribution) > len(rmd2_swh_distribution)
+        else (rmd2_swh_distribution, rmd1_swh_distribution)
+    )
+
+    compare_context_pair(
+        index_context=index_content_dict,
+        compare_context=compare_context_dict,
+        element_json_path="$.service_water_heating_distribution_systems[*]",
+        extra_schema=EXTRA_SCHEMA["ServiceWaterHeatingDistributionSystem"][
+            "Data Elements"
+        ],
+        error_msg_list=errors_list,
+        search_key=compare_context_str,
+        required_equal=True,
+    )
+
+    # service_water_heating_equipment
+    assert_(
+        len(rmd1_swh_equipment_dict.swh_heating_eq)
+        == len(rmd2_swh_equipment_dict.swh_heating_eq),
+        f"Unequal numbers of SWH Equipment between the two models for {swh_distribution_id}",
+    )
+
+    for swh_eq_id in rmd1_swh_equipment_dict.swh_heating_eq:
+        swh_eq_1 = find_exactly_one_service_water_heating_equipment(rmd1, swh_eq_id)
+        swh_eq_2 = find_exactly_one_service_water_heating_equipment(rmd2, swh_eq_id)
+
+        index_content_dict, compare_context_dict = (
+            (swh_eq_1, swh_eq_2)
+            if len(swh_eq_1) > len(swh_eq_2)
+            else (swh_eq_2, swh_eq_1)
+        )
```

Comment:
This is unnecessary. Instead, we should check whether `swh_eq_1` is None and `swh_eq_2` is None. If either is None, add to error_list and skip all the following logics.

Reply:
Addressed.